lib/dao.js

```diff
@@ -3059,7 +3059,10 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-
+          // if connector is cloudant, add the `_rev` property back before passing it to the user
+          if (connector.name === 'cloudant') {
+            context.data._rev = data._rev;
```

Comment:
Other keys should not change. This on `replace` method and if the data the user passes on to be replaced with changes back from the driver then there are our connector does not do the right job. Only `_rev` of an existing instance will be updated upon `replace`/`update`.